File: sdks/python/src/paymentflow/resources/events.py
"""``client.events`` — the event log behind webhooks (M22.6).

The same events your endpoints receive, readable after the fact. This is what a receiver
reconciles against when it suspects it missed one, so both methods are read-only: an event is a
record of something that happened and is not a thing a caller creates.
"""

from __future__ import annotations

from typing import Optional

from .._generated.models import EventResponse
from .._generated.operations import OPERATIONS
from .._pagination import CursorPage
from .._transport import RequestOptions
from ._base import Resource, query_of

__all__ = ["Events"]


class Events(Resource):
    def retrieve(self, event_id: str, *, options: Optional[RequestOptions] = None) -> EventResponse:
        """Retrieves one event."""
        result: EventResponse = self._send(OPERATIONS["getEvent"], path={"id": event_id}, options=options)
        return result

    def list(
        self,
        *,
        limit: Optional[int] = None,
        starting_after: Optional[str] = None,
        type: Optional[str] = None,
        created_after: Optional[str] = None,
        created_before: Optional[str] = None,
        options: Optional[RequestOptions] = None,
    ) -> CursorPage[EventResponse]:
        """Lists your events, most recent first.

        ``type`` filters to one event type, such as ``payment.captured``. It shadows the
        builtin inside this method and nowhere else, which is the trade the wire name is worth:
        the alternative is an ``event_type`` argument that matches neither the API nor the
        Node SDK.
        """
        query = query_of(
            limit=limit,
            starting_after=starting_after,
            type=type,
            created_after=created_after,
            created_before=created_before,
        )
        return self._list_cursor(OPERATIONS["listEvents"], query, options)

File: sdks/node/src/resources/events.ts
/**
 * `client.events` — the event log behind webhooks (M22.3).
 *
 * The same events your endpoints receive, readable after the fact. This is what a receiver
 * reconciles against when it suspects it missed one, so both methods are read-only: an event
 * is a record of something that happened and is not a thing a caller creates.
 */

import { OPERATIONS } from '../generated/operations.js';
import type { EventResponse } from '../generated/models.js';
import type { RequestOptions } from '../transport.js';
import type { CursorPage } from '../pagination.js';
import { Resource } from './base.js';

/** What `events.list` accepts. */
export type EventListParams = {
  readonly limit?: number;
  readonly starting_after?: string;
  /** Only events of this type, such as `payment.captured`. */
  readonly type?: string;
  readonly created_after?: string;
  readonly created_before?: string;
};

export class Events extends Resource {
  /** Retrieves one event. */
  retrieve(id: string, options?: RequestOptions): Promise<EventResponse> {
    return this.send<EventResponse>(OPERATIONS.getEvent, { path: { id }, options });
  }

  /** Lists your events, most recent first. The result paginates transparently. */
  list(params: EventListParams = {}, options?: RequestOptions): Promise<CursorPage<EventResponse>> {
    return this.listCursor<EventResponse>(OPERATIONS.listEvents, params, options);
  }
}
